### PR TITLE
Allow custom SMTP server

### PR DESCRIPTION
**Pull request by zmc**

### CODE REVIEW

File: .functional_ci_setup.py [hunk 1]
```diff
@@ -101,6 +101,8 @@ def get_ocsci_conf():
     )
     if env.get("DOWNSTREAM") == "true":
         conf_obj['REPORTING']['us_ds'] = 'DS'
+    if env.get("SMTP_SERVER"):
+        conf_obj['REPORTING']['email'] = dict(smtp_server=env['SMTP_SERVER'])
```
Comment (clacroix12) [inline .functional_ci_setup.py]: Won't this essentially remove `conf_obj['REPORTING']['email']['address']`?

Comment (zmc) [reply .functional_ci_setup.py]: Nope, the config system doesn't work that way (unit tests prove that)

Comment (zmc) [reply .functional_ci_setup.py]: If you want to verify:
```
from ocs_ci.framework import config

my_conf = dict(REPORTING=dict(email=dict(smtp_server='my_server')))
config.update(my_conf)
assert config.REPORTING['email']['smtp_server'] == 'my_server'
assert config.REPORTING['email']['address'] == 'ocs-ci@redhat.com'
```

Comment (clacroix12) [reply .functional_ci_setup.py]: Oh I understand now, this is for a custom config that's updating the defaults. I was thinking this was manipulating the actual config object.

**Pull request merged**